dottytest/src/test/scala/ParseDottySuite.scala

```diff
@@ -0,0 +1,158 @@
+package scala.meta.parser.dotty
+
+import scala.meta._
+import scala.meta.dialects.Dotty
+import munit.FunSuite
+
+import java.io.File
+
+import sys.process._
+import scala.language.postfixOps
+
+class ParseDottySuite extends FunSuite {
+
+  //increase this number if branch code changes :)
+  val directoryName = "dotty-codebase2"
+  // for my local testing
+  // val directoryName = "/home/kpbochenek/vl/github/kris/dotty/"
+
+  //NOTE(kbochenek): when dotty merges this switch to dotty repo/tag
+  val dottyGithubTag = "minor-syntax-fixes"
+  val dottyRepo = "https://github.com/kpbochenek/dotty.git"
+
+  def fetchDottyCodebase(): Unit = {
+    if (!scala.reflect.io.File(directoryName).exists) {
+      println("cloning dotty code to: " + ("pwd" !!))
+      s"git clone --depth 1 --branch ${dottyGithubTag} ${dottyRepo} ${directoryName}" !!
+    }
+  }
+
+  var checkedFiles = 0
+  var errors = 0
+  var error: Option[Throwable] = None
+
+  test("parse-dotty-suite") {
+    fetchDottyCodebase()
+
+    checkFilesRecursive(new File(directoryName))
+
+    println("--------------------------")
+    println(s"Files parsed correctly ${checkedFiles}")
+    println(s"Files errored: ${errors}")
+    println("--------------------------")
+    error.foreach(e => throw e)
+  }
+
+  def checkFilesRecursive(parent: File): Unit = {
+    if (ignoreParts.exists(p => parent.getAbsolutePath().contains(p))) return
+    if (parent.isDirectory()) {
+      for (f <- parent.listFiles()) checkFilesRecursive(f)
+    } else {
+      if (parent.getAbsolutePath().endsWith(".scala")) {
+        checkFile(parent)
+      }
+    }
+  }
+
+  def checkFile(file: File): Unit = {
+    if (excluded(file.getAbsolutePath)) {
+      try {
+        Input.File(file.getAbsoluteFile()).parse[Source].get.structure
+        println("File marked as error but parsed correctly " + file.getAbsolutePath())
+      } catch {
+        case e: Throwable => errors += 1
+      }
+
+    } else {
+      try {
+        Input.File(file.getAbsoluteFile()).parse[Source].get.structure
+        checkedFiles += 1
+      } catch {
+        case e: Throwable =>
+          println(s"Failed for file ${file.getAbsolutePath}")
+          println(s"Error: " + e.getMessage())
+          errors += 1
+          error = Some(e)
+      }
+    }
+  }
+
+  def excluded(path: String): Boolean = {
+    excludedList.exists(el => path.endsWith(el))
+  }
+
+  final val excludedList = List(
```

Comment:
I think it's also fine if we remove the list and just allow the test to be run separately from everything else. We could also add additional repositories at some point ot be able to properly test everything.

Reply:
You run parsing dotty codebase, 60 files are not parseable, you cannot fix everything.
You want to see if you introduced regression in your changes or not.
How to do this without exclusion list?

Reply:
Ok, maybe in that case it makes sense. We can remove more things from exclusions later on